src/prefect/orion/schemas/core.py

```diff
@@ -663,12 +662,17 @@ async def from_orm_model(
             flat_data = dict_to_flatdict(data)
             # iterate over the (possibly nested) secret fields
             # and obfuscate their data
-            for field in orm_block_document.block_schema.fields.get(
+            for secret_field in orm_block_document.block_schema.fields.get(
                 "secret_fields", []
             ):
-                key = tuple(field.split("."))
-                if flat_data.get(key) is not None:
-                    flat_data[key] = obfuscate_string(flat_data[key])
+                secret_key = tuple(secret_field.split("."))
+                if flat_data.get(secret_key) is not None:
+                    flat_data[secret_key] = obfuscate_string(flat_data[secret_key])
+                elif "*" in secret_key:
+                    wildcard_index = secret_key.index("*")
```

Comment:
Not sure if this is necessary, but what if we use `.*` to narrow it down:
```suggestion
                elif ".*" in secret_key:
                    wildcard_index = secret_key.index(".*")
```

Reply:
Also, might be nice to leave some comments on this.

Reply:
`secret_key` is a tuple created by splitting the field path on periods, so this wouldn't work. I've added some comments to shed some light on how the obfuscation in the backend works.